controller/lib/include/net_interface.h

```diff
@@ -62,12 +62,27 @@ class net_interface
     /// \return The network interface IP address by index for a device.
     ///
     AVDECC_CONTROLLER_LIB32_API virtual const char * STDCALL get_dev_ip_address_by_index(size_t dev_index, size_t ip_index) = 0;
+    
+    ///
+    /// \return The network interface MAC address by index for a device.
+    ///
+    AVDECC_CONTROLLER_LIB32_API virtual uint64_t STDCALL get_dev_mac_addr_by_index(size_t dev_index) = 0;
 
     /// This function is OS dependant. In linux and OSX the return is the same as for a call to
     /// get_dev_desc_by_index(). In Windows this function returns a GUID as a string.
     /// \return The corresponding network interface name by index.
     ///
     AVDECC_CONTROLLER_LIB32_API virtual char * STDCALL get_dev_name_by_index(size_t dev_index) = 0;
+    
+    ///
+    /// \return True if ip_addr_str is the IP address of the device specified by dev_index.
+    ///
+    AVDECC_CONTROLLER_LIB32_API virtual bool STDCALL find_selected_interface_by_ip_address(size_t dev_index, char * ip_addr_str) = 0;
+    
+    ///
+    /// \return True if mac_addr is the MAC address of the device specified by dev_index.
+    ///
+    AVDECC_CONTROLLER_LIB32_API virtual bool STDCALL find_selected_interface_by_mac_address(size_t dev_index, uint64_t mac_addr) = 0;
```

Comment:
I think "find" is the wrong word here. To me "find" implies a search operation, but that is not what this function does. Maybe compare_interfaces_mac_address() ?